src/cache.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Tran Tuan Linh
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use libc::{self, size_t};
+
+use crate::ffi;
+
+// Cache is a cache used to store data read from data in memory.
+pub struct Cache {
+    pub(crate) inner: *mut ffi::rocksdb_cache_t,
+}
+
+#[allow(dead_code)]
+impl Cache {
+    /// Create a lru cache with capacity
+    pub fn new_lru_cache(capacity: size_t) -> Cache {
+        let cache = unsafe { ffi::rocksdb_cache_create_lru(capacity) };
+        if cache.is_null() {
+            panic!("Could not create Cache");
```

Comment:
Personally I would prefer to return `Result` instead of panicking here, but I'm not 100% sure because we do have very similar code with panic for `Options`. On the other hand, `DB::open` returns a result.